bgpd/bgp_routemap.c

```diff
@@ -4122,778 +4151,1079 @@ DEFUN (no_match_evpn_route_type,
+	snprintf(xpath_value, sizeof(xpath_value),
+		 "%s/rmap-match-condition/frr-bgp-route-map:probability",
+		 xpath);
+	nb_cli_enqueue_change(vty, xpath_value, NB_OP_MODIFY,
+			      argv[idx_number]->arg);
+
+	return nb_cli_apply_changes(vty, NULL);
 }
 
 
-DEFUN (no_match_probability,
-       no_match_probability_cmd,
-       "no match probability [(1-99)]",
-       NO_STR
-       MATCH_STR
-       "Match portion of routes defined by percentage value\n"
-       "Percentage of routes\n")
+DEFUN(no_match_probability, no_match_probability_cmd,
+      "no match probability [(1-99)]",
+      NO_STR MATCH_STR
+      "Match portion of routes defined by percentage value\n"
+      "Percentage of routes\n")
 {
 	int idx_number = 3;
-	if (argc <= idx_number)
-		return bgp_route_match_delete(vty, "probability", NULL,
-					      RMAP_EVENT_MATCH_DELETED);
-	return bgp_route_match_delete(vty, "probability", argv[idx_number]->arg,
-				      RMAP_EVENT_MATCH_DELETED);
-}
+	const char *xpath =
+		"./match-condition[condition='frr-bgp-route-map:probability']";
+	char xpath_value[XPATH_MAXLEN];
 
+	nb_cli_enqueue_change(vty, xpath, NB_OP_DESTROY, NULL);
 
-DEFUN (match_ip_route_source,
-       match_ip_route_source_cmd,
-       "match ip route-source <(1-199)|(1300-2699)|WORD>",
-       MATCH_STR
-       IP_STR
-       "Match advertising source address of route\n"
-       "IP access-list number\n"
-       "IP access-list number (expanded range)\n"
-       "IP standard access-list name\n")
-{
-	int idx_acl = 3;
-	return bgp_route_match_add(vty, "ip route-source", argv[idx_acl]->arg,
-				   RMAP_EVENT_FILTER_ADDED);
+	if (argc <= idx_number)
+		return nb_cli_apply_changes(vty, NULL);
+
+	snprintf(xpath_value, sizeof(xpath_value),
+		 "%s/rmap-match-condition/frr-bgp-route-map:probability",
+		 xpath);
+	nb_cli_enqueue_change(vty, xpath_value, NB_OP_DESTROY,
+			      argv[idx_number]->arg);
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+
+DEFPY(match_ip_route_source, match_ip_route_source_cmd,
+      "match ip route-source <(1-199)$acll|(1300-2699)$aclh|WORD$name>",
+      MATCH_STR IP_STR
+      "Match advertising source address of route\n"
+      "IP access-list number\n"
+      "IP access-list number (expanded range)\n"
+      "IP standard access-list name\n")
+{
+	const char *xpath =
+		"./match-condition[condition='frr-bgp-route-map:ip-route-source']";
+	char xpath_value[XPATH_MAXLEN + 32];
+	int acln = acll ? acll : aclh;
+
+	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
+	if (name) {
+		snprintf(xpath_value, sizeof(xpath_value),
+			 "%s/rmap-match-condition/frr-bgp-route-map:list-name",
+			 xpath);
+		nb_cli_enqueue_change(vty, xpath_value, NB_OP_MODIFY, name);
+	} else /* if (acll || aclh) */ {
+		if ((acln >= 1 && acln <= 99)
+		    || (acln >= 1300 && acln <= 1999)) {
+			snprintf(
+				xpath_value, sizeof(xpath_value),
+				"%s/rmap-match-condition/frr-bgp-route-map:access-list-num",
```

Comment:
This needs an update: access list is now only a string.

- https://github.com/FRRouting/frr/blob/master/yang/frr-route-map.yang#L264
- https://github.com/FRRouting/frr/blob/master/yang/frr-filter.yang#L109

Reply:
DEFPY(match_ip_route_source, match_ip_route_source_cmd,
      "match ip route-source <(1-199)$acll|(1300-2699)$aclh|WORD$name>",
      MATCH_STR IP_STR
      "Match advertising source address of route\n"
      "IP access-list number\n"
      "IP access-list number (expanded range)\n"
      "IP standard access-list name\n")

We are maintaining 3 different types.
access-list-num for 1-99
access-list-num-extended 1300-269
list-name for word

Reply:
This one is still not addressed. As Rafael pointed out, the access-list is referenced by its name.
It doesn't matter that this name may actually be a set of characters or a number. It's still a string.
Instead of using three different leafs - `access-list-num`, `access-list-num-extended` and `access-list-name` just use single `access-list-name` for all three cases.